permission-portal-cloud-functions/functions/src/index.ts

```diff
@@ -51,38 +51,44 @@ function doesOrganizationExist(organizationID: string): Promise<boolean> {
 
 // Deletes user from users table, if an entry exists
 function usersCollectionDeleteUser(email: string) {
-  db.collection('users')
+  return db
+    .collection('users')
     .doc(email)
     .get()
     .then((user) => {
       if (user.exists) {
-        user.ref.delete().catch((err) => {
+        return user.ref.delete().catch((err) => {
           console.error(err);
+          throw err;
         });
       }
+      throw new Error('Error deleting user from users collection.');
     })
     .catch((err) => {
       console.error(err);
+      throw err;
     });
 }
 
 // Deletes user from firebase auth and from users table, if an entry exists
-function deleteUser(uid: string) {
-  auth
+function deleteUserByUid(uid: string) {
+  return auth
     .getUser(uid)
     .then((userRecord) => {
       const email = userRecord.email;
-      auth
+      return auth
         .deleteUser(uid)
         .then(() => {
-          usersCollectionDeleteUser(email ? email : '');
+          return usersCollectionDeleteUser(email ? email : '');
```

Comment:
is blank email valid? I suppose it'll hit the user in org check.

Reply:
The blank email will cause `usersCollectionDeleteUser` to throw an error when we try to get a document indexed by `''`. We could definitely do some more elegant error checking / messages but I made the judgement call that this was good enough for now, if we start hitting this then I'll add some fancier checks.